cpp/src/io/functions.cpp

```diff
@@ -436,7 +436,13 @@ void write_orc(orc_writer_options const& options)
   auto writer = std::make_unique<orc::detail::writer>(
     std::move(sinks[0]), options, io_detail::single_write_mode::YES, cudf::get_default_stream());
 
-  writer->write(options.get_table());
+  try {
+    writer->write(options.get_table());
+  } catch (const std::exception& e) {
```

Comment:
```suggestion
  } catch (std::exception const& e) {
```

Reply:
In fact, 
```suggestion
  } catch (std::exception const&) {
```
Question: Is it customary to assume it's a `std::exception`? Would `catch(...)` be gauche?

Reply:
`catch(...)` should be good enough, since we don't care about what exception is.